Development/src/components/makeConnection.js

```diff
@@ -120,94 +203,15 @@ const makeConnection = (senderID, receiverID, endpoint) => {
-                                'destination_host'
-                            );
-                            setJSON.set(
-                                patchData,
-                                `/$staged/transport_params/${index}/source_host`,
-                                destination_host,
-                                true
-                            );
-                            const destination_port = get(
-                                leg,
-                                'destination_port'
-                            );
-                            setJSON.set(
-                                patchData,
-                                `/$staged/transport_params/${index}/source_port`,
-                                destination_port,
-                                true
-                            );
-                        });
-                        break;
-                    case 'urn:x-nmos:transport:rtp':
-                        senderTransportParams.forEach(function(leg, index) {
-                            const destination_ip = get(leg, 'destination_ip');
-                            if (isMulticast(destination_ip)) {
-                                setJSON.set(
-                                    patchData,
-                                    `/$staged/transport_params/${index}/multicast_ip`,
-                                    destination_ip,
-                                    true
-                                );
-                            } else {
-                                setJSON.set(
-                                    patchData,
-                                    `/$staged/transport_params/${index}/interface_ip`,
-                                    destination_ip,
-                                    true
-                                );
-                            }
-                        });
-                        break;
-                    case 'urn:x-nmos:transport:websocket':
-                        copyTransportParams(
-                            senderTransportParams,
-                            getExtParams(
-                                get(data.sender, '$active.transport_params')
-                            ),
-                            patchData
-                        );
-                        break;
-                    default:
-                }
                 return {
                     id: get(data.receiver, 'id'),
                     data: patchData,
                     previousData: data.receiver,
                 };
             })
-            .then(params =>
-                dataProvider('UPDATE', 'receivers', params, {
-                    onSuccess: { refresh: true },
-                })
-            )
+            .then(params => dataProvider('UPDATE', 'receivers', params))
             .then(() => resolve())
-            .catch(error => reject(error.message));
+            .catch(error => reject(error));
```

Comment:
Hmm, where has the use of `message` ended up?

Reply:
As the error may be network related and would therefore have an error.debug field, I wanted to wait until we handle the error before using the string form. The error class additionally has a toString() method which we should be using.

Reply:
Sure, but in other places I can only see handling of the NMOS form with `debug`?

Reply:
Apologies but I don't quite follow, where specifically are you referencing?
An oddity with the format of errors is they may not always have a debug field. Unfortunately using the error.toString() method does not include the debug message in the returned string. This is why I have chosen to handle errors with a debug property separately in some locations, as the debug property includes useful information regarding schema validation etc..

Reply:
All the error handling I could see, assumes the NMOS JSON code/message/debug props. So when an error doesn't have those things, what happens? Maybe I am looking at two different levels of error handling.

Reply:
Currently if the error does not have the body property containing the debug information, I am just using error.toString()
```javascript
.catch(error => {
    if (error && error.hasOwnProperty('body'))
        notify(
            get(error.body, 'error') +
                ' - ' +
                get(error.body, 'code') +
                ' - ' +
                get(error.body, 'debug'),
            'warning'
        );
    notify(error.toString(), 'warning');
});
```
I believe the ideal solution to this problem would be to rewrite the data provider layer to use our own fetch instead of the react-admin fetchJson. This way we could create our HttpError class with a toString() method that returns the string we would like. Is this work you think would be good to undertake today?

Reply:
No :-)
That makes sense. Thanks!